epengine/models/inference.py

```diff
@@ -1807,6 +2353,297 @@ def Open(cls, path: Path) -> "RetrofitCosts":
 
 RETROFIT_COST_CACHE: dict[Path, RetrofitCosts] = {}
 
+
+class IncentiveFactor(ABC):
+    """An abstract base class for all incentive factors."""
+
+    @abstractmethod
+    def compute(self, features: pd.DataFrame, costs_df: pd.DataFrame) -> pd.Series:
+        """Compute the incentive for a given feature and cost."""
+        pass
+
+
+class FixedIncentive(BaseModel, IncentiveFactor):
+    """A fixed incentive amount."""
+
+    amount: float
+    error_scale: float | None = Field(
+        ...,
+        description="The expected error of the incentive estimate, if there are associated ranges due to regions, contractors, etc.",
+        ge=0,
+        le=1,
+    )
+    units: Literal["USD"]
+    description: str = Field(
+        ...,
+        description="A description of the fixed incentive.",
+    )
+    source: str = Field(
+        ...,
+        description="The source of the fixed incentive.",
+    )
+
+    def compute(self, features: pd.DataFrame, costs_df: pd.DataFrame) -> pd.Series:
+        """Compute the incentive for a given feature."""
+        if self.error_scale is None:
+            return pd.Series(
+                np.full(len(features), self.amount),
+                index=features.index,
+            )
+        else:
+            return pd.Series(
+                np.random.normal(
+                    self.amount, self.amount * self.error_scale, len(features)
+                ).clip(min=0),
+                index=features.index,
+            )
+
+
+class VariableIncentive(BaseModel, IncentiveFactor):
```

Comment:
Unclear to me why this needs its own class - I *think* it is the same pattern as the original VariableCost class, which probably could just be given a different name.